File: src/main/java/com/matag/game/turn/phases/Phase.java
package com.matag.game.turn.phases;

import com.matag.game.status.GameStatus;

public interface Phase {

  String getName();

  void action(GameStatus gameStatus);

  Phase getNextPhase(GameStatus gameStatus);

  void set(GameStatus gameStatus);

  void next(GameStatus gameStatus);
}

File: src/main/kotlin/com/matag/game/turn/phases/Phase.kt
package com.matag.game.turn.phases

import com.matag.game.status.GameStatus

interface Phase {
    val name: String

    fun action(gameStatus: GameStatus)

    fun getNextPhase(gameStatus: GameStatus): Phase

    fun set(gameStatus: GameStatus)

    fun next(gameStatus: GameStatus)
}
